packages/core-v2-sdk/src/models/Token.ts

```diff
@@ -54,4 +55,15 @@ export class Token {
   getBalanceOf(address: string): Promise<string> {
     return this.dataSource.getBalanceOf(address);
   }
+
+  /**
+   * Sets approval of token access up to a certain amount
+   * @param {Signer} signer - Signer.
+   * @param {string} who - Address to approve access to.
+   * @param {string} [amount] - Amount approved for, defaults to maximum.
+   * @return {Promise<boolean>} successful - Boolean denoting a successful approval.
+   */
+  approve(signer: Signer, who: string, amount?: string): Promise<boolean> {
```

Comment:
Minor nit: let's change it from `who` to `spender` to match the erc20 spec: https://ethereum.org/en/developers/docs/standards/tokens/erc-20/#methods